src/firefly/plans/xafs_scan.py

```diff
@@ -151,6 +187,39 @@ def update_wavenumber_energy(self):
             else:
                 self.update_line_edit_value(line_edit, func)
 
+    def update_total_time(self):
+        try:
+            start = round(float(self.start_line_edit.text()), float_accuracy)
+            stop = round(float(self.stop_line_edit.text()), float_accuracy)
+            step = round(float(self.step_line_edit.text()), float_accuracy)
+            weight = self.weight_spinbox.value()
+            exposure_time = self.exposure_time_spinbox.value()
+            if self.k_space_checkbox.isChecked():
+                self.kErange = KRange(
+                    k_min=start,
+                    k_max=stop,
+                    k_step=step,
+                    k_weight=weight,
+                    exposure=exposure_time,
+                )
+
+            else:
+                self.kErange = ERange(
+                    E_min=start,
+                    E_max=stop,
+                    E_step=step,
+                    weight=weight,
+                    exposure=exposure_time,
+                )
+
+        # when there's invalid number in the lineEdits, set an empty kErange
+        except:
```

Comment:
Avoid a bare except statement, catch the specific exception you're interested in.

Reply:
solved: `except ValueError`